common/sagas/transactions.ts

```diff
@@ -34,6 +49,38 @@ export function* fetchTxData(action: FetchTransactionDataAction): SagaIterator {
   yield put(setTransactionData({ txhash, data, receipt, error }));
 }
 
+export function* saveBroadcastedTx(action: BroadcastTransactionQueuedAction) {
+  const txBuffer = action.payload.serializedTransaction;
+  const txIdx = action.payload.indexingHash;
+
+  const res: BroadcastTransactionSucceededAction | BroadcastTransactionFailedAction = yield take([
+    TxTypeKeys.BROADCAST_TRANSACTION_SUCCEEDED,
+    TxTypeKeys.BROADCAST_TRASACTION_FAILED
+  ]);
+
+  // If our TX succeeded, save it and update the store.
+  if (
+    res.type === TxTypeKeys.BROADCAST_TRANSACTION_SUCCEEDED &&
+    res.payload.indexingHash === txIdx
+  ) {
+    const tx = new EthTx(txBuffer);
+    saveRecentTransaction(res.payload.broadcastedHash, tx);
+    yield call(loadRecentTxs);
```

Comment:
Instead of re-loading the all tx's again to rehydrate the redux store, this should just dispatch an action to the reducer `TRANSACTIONS_SET_RECENT_TRANSACTIONS`